### PR TITLE
Added definitions for crpc

### PR DESCRIPTION
**Pull request by 0xdeafcafe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

The `no-unnecessary-generics` rule was thrown, however the generic is sort of needed in this function, so I updated the lint file to ignore it. Is this allowed?

### CODE REVIEW

**Comment by 0xdeafcafe** [comment]
I assume the travis failure was due to GitHub webhooks being down, rather than a bug with the code?

**Comment by 0xdeafcafe** [comment]
What? I was waiting for a response from a maintainer.. @RyanCavanaugh can you re-open this?

File: types/crpc/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for crpc 0.2
+// Project: https://github.com/billinghamj/crpc
+// Definitions by: Alexander Forbes-Reed <https://github.com/0xdeafcafe>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.9
+
+export type Client = <TReq, TRes>(path: string, body: TReq, options?: {} | null) => Promise<TRes>;
+
+export default function crpc(baseUrl: string, options?: {}): Client;
```
Comment (DanielRosenwasser) [inline types/crpc/index.d.ts]: This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`.

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

File: types/crpc/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
```
Comment (DanielRosenwasser) [inline types/crpc/tsconfig.json]: The CI is complaining because you didn't enable `strictFunctionTypes`.

File: types/crpc/tslint.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+	"extends": "dtslint/dt.json",
+	"rules": {
+		"no-unnecessary-generics": false
```
Comment (DanielRosenwasser) [inline types/crpc/tslint.json]: That `TRes` is really just a way of achieving a type assertion. I'd recommend making the return type `Promise<unknown>`so users are aware of the explicit cast.

**Review by DanielRosenwasser — COMMENTED** [review]
Please disregard our annoyingly aggressive typescript-bot and address the feedback I've provided. After that I think we should be good to merge :+1:`.

File: types/crpc/tslint.json (hunk 3 above)
Comment (0xdeafcafe) [reply types/crpc/tslint.json]: The intent of this function is to retrieve the JSON payload from an API and assert it to the type provided. So this behaviour isn't unexpected. Removing TReq, and replacing with `any` does make sense, but I think it goes against the grain to do unknown, then do the type assertion on the client side (as I feel this just adds more code).

File: types/crpc/tsconfig.json (hunk 2 above)
Comment (0xdeafcafe) [reply types/crpc/tsconfig.json]: Handled this, will open new PR with change.

**Comment by DanielRosenwasser** [comment]
Thanks!

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**